### ISSUE TITLE
as a user, I can browse audit results in a tabular form, with links to PDF and SF-SAC summaries

### ISSUE THREAD

**Issue by jadudm** [GSA-TTS/FAC#1233]
See #1170 for context.

The result of MVP search is a table, and columns should be sortable.

The result set should include:

* PDF link
* Summary link
* Auditee UEI
* Cognizant agency
* Auditee name
* State (abbrev)
* Accepted date

```
┌─────┬──────┬───────────────┬────┬───────────────┬────┬──────────────┐
│ PDF │ Form │ EJLRWNCJTJF5  │ 47 │ Berea College │ KY │ Oct 16, 2008 │
└─────┴──────┴───────────────┴────┴───────────────┴────┴──────────────┘
```

We'll show 20 rows at a time. The frontend will have the complete result set, and therefore sorting operations will not be on the 20-item view, but instead on all of the items that came back. As of this ticket, result sets will be limited to 1000 items, so there will never be more than 1000 rows available. 

The PDF and Form links can be prettied up as some kind of little lozenge if we want. Otherwise, just small hyperlinks is fine.

The PDF link will target `_blank` and force a download of the PDF. The URL format needs to be decided; it always needs to hit `app.fac.gov/pdf` (or similar), and the UID for the PDF is then checked in terms of whether or not the user can/should be able to download it. (Right now, we only have public data, so this is not a big issue, but we should prepare the pathway for when we have protected data.)

The `Form` link should target a new tab, and it should open a page where we render the SF-SAC contents through a template. In a future world, we have all of the SF-SAC contents statically rendered to our site, and we just link to the SF-SAC that is already dumped/rendered.

<strike>Clicking on the UEI launches a new search for that UEI. If possible, we treat it as a JSON POST to the search backend with only the UEI array populated with a single value.</strike> *We will move browsing of data to post-MVP.*

<strike>Clicking on the cognizant agency launches a new search for everything from that cognizant agency. We treat it as a POST with only the cog agency populated.</strike> *We will move browsing of data to post-MVP.*

No other values are hyperlinked.

To be captured elsewhere: **the backend should always return the search query that it was given along with the result set**. This will allow  the frontend to (optionally) repopulate the search fields when rendering the results. Point being, what was submitted should come back as part of the result object.

```[tasklist]
### Tasks
- [ ] Result sets come back containing the fields described as an array of JSON objects
- [ ] The result set is rendered 20 rows at a time as a USDWS sortable table
- [ ] Sort is enabled on the content columns
- [ ] Link for PDF targets download, and routes through `app.fac.gov/pdf`
- [ ] Link for Form targets a Django view that renders the entire SF-SAC through a template
- [ ] UEI is hyperlinked, and triggers a new search for just that UEI
- [ ] Cognizant Agency is hyperlinked,  and triggers a new search for just that cog agency
```

**Comment by Carley-Sullivan** [comment]
Thinking as an ARO - the following feature's placement wouldn't make sense to me:

_Clicking on the cognizant agency launches a new search for everything from that cognizant agency. We treat it as a POST with only the cog agency populated._  

As a user - when I click the cog agency link from results grid, I would expect to see the cognizant/oversight assignments for the auditee ranging back as far as possible.  If that feature isn't available for MVP, just display the Cog Assignment for that submission. 

If an ARO wants to see all records for a set of Cog/Oversight agencies, it would make sense to have it as something they can pull up from the main search screen, not results grid > new cog/over search.

I'm happy to talk about this in more detail if needed :)

**Comment by jadudm** [comment]
It makes sense. Let me explore it.

First, if they want to search by cog agency only, they can. So, that's available "at the top level." They could search by cog agency and a date range, even. Or, cog agency and a UEI. So, really, any combination of the fields we're providing.

Second, your question raises a difficult question. Lets say I searched for a date range only. (Is this a "realistic" search from an ARO perspective? Perhaps not. But, it *has to work* if we allow it.) I see a whole bunch of results. And, we decide to hyperlink the results in the cognizant agency part of the table. 

At this point, we are moving from *search* to *browse*.

The context I have for this search (at the moment I render the page) is a date range. There is no UEI or notion of an auditee. So, if I click on the cognizant agency... what should I do? I would claim we can do one of three (squint) things, and have it "make sense":

1. We can do a further search, and do that search on the cognizant agency *only*. Why? Because that is what the user clicked. They clicked the number 42 in the cognizant agency column, and we expand it to all grants where cognizant agency is 42.
2. We can do a further search, and do that search with all of the current parameters, but set cognizant agency to the value I click on. In this regard, we keep the date range that I entered before, and now do a search for that date range *and* cognizant agency 42.
3. We don't allow for browsing.

I'm actually starting to wonder if browsing the result set is not part of MVP. That is, if hyperlinking anything in the result set is too much for a 2-week search implementation.

I'm going to update the ticket to reflect this. @heymatthenry , @jperson1 , @timoballard , note that this just got simpler.

If we can do browse before MVP, we'll revisit it. Otherwise, it will be an exciting new feature that users will be excited about post-MVP.
